modules/copy-post.php

```diff
@@ -0,0 +1,190 @@
+            return;
+        }
+
+        if ( ! empty( $_GET[ 'copy' ] ) && 'post-new.php' === $GLOBALS[ 'pagenow' ] ) {
+            add_action( 'wp_insert_post', array( $this, 'update_post_data' ), 10, 3 );
+        }
+    }
+
+    /**
+     * Update the new (target) post data with the source post data.
+     *
+     * @param int     $target_post_id Target post ID.
+	 * @param WP_Post $post           Target post object (not used).
+	 * @param bool    $update         Whether this is an existing post being updated or not.
+     * @return void
+     */
+    function update_post_data( $target_post_id, $post, $update ) {
+        // This avoids infinite loops of trying to update our updated post.
+        if ( $update ) {
+            return;
+        }
+
+        $source_post = get_post( $_GET['copy'] );
+        if ( ! $source_post || ! $this->user_can_edit_post( $source_post ) ) {
+            return;
+        }
+
+        $update_content = $this->update_content_and_taxonomies( $source_post, $target_post_id );
+        $update_featured_image = $this->update_featured_image( $source_post, $target_post_id );
+        $update_post_format = $this->update_post_format( $source_post, $target_post_id );
+
+        // Required to satify get_default_post_to_edit(), which has these filters after post creation.
+        add_filter( 'default_title', array( $this, 'filter_title' ), 10, 2 );
+        add_filter( 'default_content', array( $this, 'filter_content' ), 10, 2 );
+        add_filter( 'default_excerpt', array( $this, 'filter_excerpt' ), 10, 2 );
+
+        do_action( 'jetpack_copy_post', $source_post, $target_post_id, $update_content, $update_featured_image, $update_post_format );
+    }
+
+    /**
+     * Determine if the current user has access to the source post.
+     *
+	 * @param WP_Post $post Source post object (the post being copied).
+     * @return bool         True if current user is the post author, or has permissions for `edit_others_posts`; false otherwise.
+     */
+    protected function user_can_edit_post( $post ) {
+        return get_current_user_id() === (int) $post->post_author || current_user_can( 'edit_others_posts' );
+    }
+
+    /**
+     * Update the target post's title, content, excerpt, categories, and tags.
+     *
+	 * @param WP_Post $source_post Post object to be copied.
+     * @param int     $target_post_id Target post ID.
+     * @return int    0 on failure, or the updated post ID on success.
+     */
+    protected function update_content_and_taxonomies( $source_post, $target_post_id ) {
+        $data = apply_filters( 'jetpack_copy_post_data', array(
+            'ID' => $target_post_id,
+            'post_title' => $source_post->post_title,
+            'post_content' => $source_post->post_content,
+            'post_excerpt' => $source_post->post_excerpt,
+            'post_category' => $source_post->post_category,
+            'tags_input' => $source_post->tags_input,
```

Comment:
Should we also attempt to copy things like the status of our Sharing and Like buttons (on or off), as well as the status of comments and trackbacks (on or off)? I think it would be a nice addition.

I could see the post format being something we could copy as well, although I assume it's not used as much.

Reply:
Added comments, pingbacks, Likes, and Sharing 👍